src/appleseed/foundation/meta/tests/test_imagefilewriter.cpp

```diff
@@ -0,0 +1,200 @@
+        std::unique_ptr<Image> image(reader.read(image_file_path.c_str()));
+            
+        for (std::size_t y = 0; y < image_size; ++y)
+        {
+            for (std::size_t x = 0; x < image_size; ++x)
+            {
+                Color4b c;
+                image->get_pixel(x, y, c);
+                EXPECT_EQ(fill_color, c);
+            }
+        }
+    }
+    
+    TEST_CASE(WriteEXR_CorrectlyWritesImagePixels)
+    {
+        const std::string image_file_path = "unit tests/outputs/test_imagefilewriter.exr";
+        const std::size_t image_size = 16;
+        const Color4b fill_color(50, 100, 150, 42);
+                        
+        write_image<Color4b>(fill_color, PixelFormatUInt8, image_size, image_file_path);
+
+        GenericImageFileReader reader;
+        std::unique_ptr<Image> image(reader.read(image_file_path.c_str()));
+            
+        for (std::size_t y = 0; y < image_size; ++y)
+        {
+            for (std::size_t x = 0; x < image_size; ++x)
+            {
+                Color4b c;
+                image->get_pixel(x, y, c);
+                EXPECT_EQ(fill_color, c);
+            }
+        }
+    }
+    
+    TEST_CASE(WriteHDR_CorrectlyWritesImagePixels)
+    {
+        const std::string image_file_path = "unit tests/outputs/test_imagefilewriter.hdr";
+        const std::size_t image_size = 16;
+        const Color3b fill_color(50, 100, 150);
+                
+        write_image<Color3b>(fill_color, PixelFormatUInt8, image_size, image_file_path);
+
+        GenericImageFileReader reader;
+        std::unique_ptr<Image> image(reader.read(image_file_path.c_str()));
+            
+        for (std::size_t y = 0; y < image_size; ++y)
+        {
+            for (std::size_t x = 0; x < image_size; ++x)
+            {
+                Color3b c;
+                image->get_pixel(x, y, c);
+                EXPECT_EQ(fill_color, c);
+            }
+        }
+    }
+    
+    TEST_CASE(WriteJPG_ApproximatelyWritesImagePixels)
+    {
+        const std::string image_file_path = "unit tests/outputs/test_imagefilewriter.jpg";
+        const std::size_t image_size = 16;
+        const Color3b fill_color(50, 100, 150);
+                
+        write_image<Color3b>(fill_color, PixelFormatUInt8, image_size, image_file_path);
+
+        GenericImageFileReader reader;
+        std::unique_ptr<Image> image(reader.read(image_file_path.c_str()));
+            
+        for (std::size_t y = 0; y < image_size; ++y)
+        {
+            for (std::size_t x = 0; x < image_size; ++x)
+            {
+                Color3b c;
+                image->get_pixel(x, y, c);
+                
+                // the actual written and expected pixel value may be off by 1 in JPG format
+                for (std::size_t i = 0; i < 3; ++i)
+                {
+                    int difference = std::min(c[i]-fill_color[i], fill_color[i] - c[i]);
+                    EXPECT_LT(difference, 2);
```

Comment:
This should be

    EXPECT_LT(2, difference);

Right now you're checking that `2 < difference`.